biom/tests/test_table.py

```diff
@@ -1815,6 +1821,196 @@ def test_add_sample_metadata_two_entries(self):
+        self.assertEqual(res_table.descriptive_equality(exp_table),
+                         'Tables appear equal')
+
+    def test_align_to_dataframe_samples_no_common_ids(self):
+        table = Table(np.array([[0, 0, 1, 0],
+                                [2, 2, 4, 0],
+                                [5, 5, 3, 0],
+                                [0, 0, 0, 1]]),
+                      ['s1', 's2', 's3', 's4'],
+                      ['o1', 'o2', 'o3', 'o4'])
+        metadata = pd.DataFrame([['a', 'control'],
+                                 ['c', 'diseased'],
+                                 ['b', 'control']],
+                                index=['s1', 's3', 's2'],
+                                columns=['Barcode', 'Treatment'])
+        with self.assertRaises(TableException):
+            table.align_to_dataframe(metadata)
+
+    @pytest.mark.skipif(not HAVE_SKBIO, reason="skbio not installed")
+    def test_align_tree_intersect_tips(self):
+        # there are less tree tips than observations
+        table = Table(np.array([[0, 0, 1, 1],
+                                [2, 3, 4, 4],
+                                [5, 5, 3, 3],
+                                [0, 0, 0, 1]]).T,
+                      ['a', 'b', 'c', 'd'],
+                      ['s1', 's2', 's3', 's4'])
+        tree = skbio.TreeNode.read([u"((a,b)f,d)r;"])
+        exp_table = Table(np.array([[0, 0, 1],
+                                    [2, 3, 4],
+                                    [5, 5, 3],
+                                    [0, 0, 1]]).T,
+                          ['a', 'b', 'd'],
+                          ['s1', 's2', 's3', 's4'])
+        exp_tree = tree
+        res_table, res_tree = table.align_tree(tree)
+        self.assertEqual(res_table.descriptive_equality(exp_table),
+                         'Tables appear equal')
+        self.assertEqual(str(exp_tree), str(res_tree))
+
+    @pytest.mark.skipif(not HAVE_SKBIO, reason="skbio not installed")
+    def test_align_tree_intersect_obs(self):
+        # table has less observations than tree tips
+        table = Table(np.array([[0, 0, 1],
+                                [2, 3, 4],
+                                [5, 5, 3],
+                                [0, 0, 1]]).T,
+                      ['a', 'b', 'd'],
+                      ['s1', 's2', 's3', 's4'])
+        tree = skbio.TreeNode.read([u"(((a,b)f, c),d)r;"])
+        exp_table = Table(np.array([[1, 0, 0],
+                                    [4, 2, 3],
+                                    [3, 5, 5],
+                                    [1, 0, 0]]).T,
+                          ['d', 'a', 'b'],
+                          ['s1', 's2', 's3', 's4'])
+        exp_tree = skbio.TreeNode.read([u"(d,(a,b)f)r;"])
+        res_table, res_tree = table.align_tree(tree)
+        self.assertEqual(res_table.descriptive_equality(exp_table),
+                         'Tables appear equal')
+        self.assertEqual(str(exp_tree), str(res_tree))
```

Comment:
Why the newick test vs `self.assertEqual(exp_tree.compare_rfd(res_tree), 0)`?

Reply:
because I'm not too familiar with Robbins Fould's distance - agree this is a better test